src/controllers/post.js

```diff
@@ -404,3 +404,104 @@ export const deleteCommentLike = async (req, res) => {
     }
   })
 }
+
+export const updateComment = async (req, res) => {
+  const id = Number(req.params.commentId)
+  const { content } = req.body
+
+  if (!content) {
+    return sendMessageResponse(res, 400, 'Must provide content')
+  }
+
+  const foundComment = await dbClient.comment.findUnique({
+    where: { id },
+    include: { user: true }
+  })
+
+  if (!foundComment) {
+    return sendMessageResponse(
+      res,
+      404,
+      'The comment with the provided id does not exist'
+    )
+  }
+
+  if (foundComment.user.id !== req.user.id) {
+    return sendMessageResponse(
+      res,
+      403,
+      'Only the comment author can edit the comment'
+    )
+  }
+
+  const updatedComment = await dbClient.comment.update({
+    where: { id },
+    data: { content },
+    include: {
+      user: true,
+      likes: {
+        include: {
+          user: {
+            select: {
+              email: true,
+              id: true,
+              cohortId: true,
+              role: true,
+              profile: true
+            }
+          }
+        }
+      }
+    }
+  })
+
+  return sendDataResponse(res, 200, updatedComment)
+}
+
+export const deleteComment = async (req, res) => {
+  const id = Number(req.params.commentId)
+
+  const foundComment = await dbClient.comment.findUnique({
+    where: { id },
+    include: { user: true, post: { include: { user: true } } }
+  })
+
+  if (!foundComment) {
+    return sendMessageResponse(
+      res,
+      404,
+      'The comment with the provided id does not exist'
+    )
+  }
+
+  if (
+    req.user.role === 'TEACHER' ||
+    req.user.role === 'ADMIN' ||
+    foundComment.user.id === req.user.id ||
+    foundComment.post.user.id === req.user.id
+  ) {
```

Comment:
This is quite a big condition. Let's move it to a function so we can give this check a readable name, make it easier for other devs to know what the condition is